components/table/SearchFilterSort.tsx

```diff
@@ -38,6 +46,38 @@ export default function SearchFilterSort() {
     },
     [searchParams, replace, pathname],
   );
+  console.log(new URLSearchParams(searchParams?.toString()).toString());
+  const handleDownload = async () => {
+    const params = new URLSearchParams(searchParams?.toString());
+
+    try {
+      // Fetch the data from the API
+      const response = await fetch(`/api/${pathname.split('/')[1]}/allprojects?${params.toString()}`);
```

Comment:
I still believe `pathname.split('/')[1]` cause unexpected results in the future. let's bring another prop to indicate the page is for `private-cloud` or `public-cloud`.